src/Platforms/AbstractPlatform.php

```diff
@@ -1973,6 +2003,14 @@ public function supportsColumnCollation(): bool
         return false;
     }
 
+    /**
+     * A flag that indicates whether the platform supports functional indexes.
+     */
+    public function supportsFunctionalIndex(): bool
+    {
+        return true;
+    }
```

Comment:
Repeating myself: I don't think we need this `supports()` function. We've removed a lot of those and I'm not eager to add any new ones.

Reply:
Why we have `supportsPartialIndexes` but can't have `supportsFunctionalIndex` ?
What's the difference ?

Reply:
The difference is that the method in question was introduced in 2014 and we have 2024 now. Quoting myself:

> We've removed a lot of those and I'm not eager to add any new ones.

Reply:
I can't figure out what the alternative is? If you told me this approach is wrong, what is the right approach?
There are databases that do not supports functional indexes, so we must have a way to check and know that.

Reply:
> I can't figure out what the alternative is?

Create a new method in the platform classes that renders DDL for a functional index. Let it throw by default. Override that method inside the platform classes that support functional indexes.

> There are databases that do not supports functional indexes, so we must have a way to check and know that.

No, we don't.